cms/bundles/utils.py

```diff
@@ -25,6 +28,67 @@
     from .models import Bundle
 
 
+@dataclass
+class BundleAPIBundleMetadata:
+    """BundleAPIBundleMetadata is the core dataclass that normalizes bundle metadata
+    for comparison between CMS and Bundle API.
+
+    It normalizes:
+    - preview_teams: None -> [], sorted by id for consistent comparison
+    - scheduled_at: datetime/string -> UTC ISO-8601 format (no microseconds)
+    - bundle_type: Auto-set to SCHEDULED if scheduled_at present, else MANUAL
+    - managed_by: Defaults to WAGTAIL
+
+    This normalization ensures that equivalent bundle data from different sources
+    (CMS Bundle, Bundle API JSON) can be reliably compared for sync decisions.
+    """
+
+    title: str | None = None
+    bundle_type: Literal["MANUAL", "SCHEDULED"] | None = None
+    state: str | None = None
+    managed_by: Literal["WAGTAIL"] | None = None
+    preview_teams: list[dict[Literal["id"], str]] | None = None
+    scheduled_at: datetime | str | None = None
+
+    def __post_init__(self) -> None:
+        self.preview_teams = sorted(
+            self.preview_teams or [],
+            key=lambda team: team.get("id", ""),  # Ensure consistent ordering for equality checks
+        )
+        self.scheduled_at = _normalise_date(self.scheduled_at)
+        if not self.bundle_type:
+            self.bundle_type = "SCHEDULED" if self.scheduled_at else "MANUAL"
+
+        if not self.managed_by:
+            self.managed_by = "WAGTAIL"
+
+    def as_dict(self) -> dict:
+        return asdict(self)
+
+    @classmethod
+    def from_bundle(cls, bundle: "Bundle") -> "BundleAPIBundleMetadata":
+        """Return a BundleAPIBundleMetadata instance populated from a Bundle instance."""
+        return cls(
+            title=bundle.name,
+            state=str(bundle.status),
+            managed_by="WAGTAIL",
+            preview_teams=_get_preview_teams_for_bundle(bundle),
+            scheduled_at=bundle.scheduled_publication_date,
+        )
+
+    @classmethod
+    def from_api_response(cls, api_response: dict[str, Any]) -> "BundleAPIBundleMetadata":
```

Comment:
This wasn't mentioned in the PR description, but `from_api_response` is currently not used. Is this for future use?